src/OrchardCore.Modules/OrchardCore.Settings/Startup.cs

```diff
@@ -93,6 +93,8 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddScoped<IDisplayDriver<DeploymentStep>, SiteSettingsDeploymentStepDriver>();
 
             services.AddScoped<IRecipeEnvironmentProvider, RecipeEnvironmentSiteNameProvider>();
+
+            services.AddTransient<IConfigureOptions<ResourceOptions>, ResourceConfigureOptions>();
```

Comment:
Creating ResourceOptions per request is eanough - use Scoped

Reply:
I don’t think this is permitted as IConfigurationOption is resolved by singleton service.

Reply:
As expected, changing it causes the following exception

```
InvalidOperationException: Cannot consume scoped service 'System.Collections.Generic.IEnumerable`1[Microsoft.Extensions.Options.IConfigureOptions`1[OrchardCore.ResourceManagement.ResourceOptions]]' from singleton 'Microsoft.Extensions.Options.IOptions`1[OrchardCore.ResourceManagement.ResourceOptions]'.
```

Reply:
You  can’t consume low lifespan in higher lifespan services. If it it’s consumed by singleton then transient will not help here either,

which singleton service use this? I couldn’t find it in the code

Reply:
IOptions is singleton so anytime you register `IConfigureOptions<>` it has to be singleton. Check the error I supplied above.

Reply:
> it has to be singleton

Or transient ;)

As `IConfigureOptions<ResourceManagementOptions>` is.

Both could be singletons, but better I think to keep them transients allowing the options to be re-computed, e.g. if site settings change in a multi nodes env, even if normally in that case the tenant is reloaded locally and then on each node if the distributed tenants feature is enabled.

Reply:
@jtkech agreed!